lms/djangoapps/discussion_api/tests/test_views.py

```diff
@@ -866,6 +896,31 @@ def test_pagination(self):
             }
         )
 
+    @ddt.data(
+        (True, "endorsed_comment"),
+        ("true", "endorsed_comment"),
+        ("1", "endorsed_comment"),
+        (False, "non_endorsed_comment"),
+        ("false", "non_endorsed_comment"),
+        ("0", "non_endorsed_comment"),
+    )
```

Comment:
@jcdyer actually NullBooleanField's uses widget as NullBooleanSelect and that widget does not handle "true" or "1" and "false" or "0" and defaults to None for any other value than True and False, so I had to override the widget and define and use explicit choices.